test/Unit/Durable/DurableTimerTests.cs

```diff
@@ -0,0 +1,260 @@
+            );
+            OrchestrationContext context = new OrchestrationContext { History = history, CurrentUtcDateTime = startTime };
+            _orchestrationBindingInfo = new OrchestrationBindingInfo("ContextParameterName", context);
+            if (!timerScheduledAndFired)
+            {
+                EmulateStop(_durableTimer);
+            }
+
+            _durableTimer.CreateTimerAndStop_OrContinue(context:context, fireAt: fireAt);
+
+            if (timerScheduledAndFired)
+            {
+                Assert.Equal(restartTime, context.CurrentUtcDateTime);
+            }
+            else
+            {
+                Assert.Equal(startTime, context.CurrentUtcDateTime);
+            }
+        }
+
+        private HistoryEvent[] CreateTimerFiredHistory(bool timerScheduledAndFired, DateTime fireAt, DateTime restartTime, bool orchestratorStartedIsProcessed)
+        {
+            var history = new List<HistoryEvent>();
+
+            if (timerScheduledAndFired)
+            {
+                int timerCreatedEventId = GetUniqueEventId();
+                history.Add(
+                    new HistoryEvent
+                    {
+                        EventType = HistoryEventType.TimerCreated,
+                        EventId = timerCreatedEventId,
+                        FireAt = fireAt
+                    }
+                );
+
+                int orchestratorStartEventId = GetUniqueEventId();
+                history.Add(
+                    new HistoryEvent
+                    {
+                        EventType = HistoryEventType.OrchestratorStarted,
+                        EventId = orchestratorStartEventId,
+                        Timestamp = restartTime,
+                        IsProcessed = orchestratorStartedIsProcessed
+                    }
+                );
+
+                history.Add(
+                    new HistoryEvent
+                    {
+                        EventType = HistoryEventType.TimerFired,
+                        EventId = GetUniqueEventId(),
+                        TimerId = timerCreatedEventId,
+                        FireAt = fireAt
+                    }
+                );
+            }
+
+            return history.ToArray();
+        }
+
+        private HistoryEvent[] CreateOrchestratorStartedHistory(DateTime date, bool isProcessed)
+        {
+            var history = new List<HistoryEvent>();
+            
+            int orchestratorStartEventId = GetUniqueEventId();
+
+            history.Add(
+                new HistoryEvent
+                {
+                    EventType = HistoryEventType.OrchestratorStarted,
+                    EventId = orchestratorStartEventId,
+                    Timestamp = date,
+                    IsProcessed = isProcessed
+                }
+            );
+
+            return history.ToArray();
+        }
+
+        private static HistoryEvent[] MergeHistories(params HistoryEvent[][] histories)
```

Comment:
We have multiple copies of some methods like this across multiple Durable unit test classes. Please extract them to a class (something like `DurableTestUtilities` - or invent a better name if you can).